### ISSUE TITLE
Migrate all JS code related with organizations to the new Builder system

### ISSUE THREAD

**Issue by xavijam** [CartoDB/cartodb#11007]
In order to start using the same Builder asset modal in Bulider and in Organization pages (https://github.com/CartoDB/cartodb/issues/11009), we should migrate all the current code to the Builder one, it could imply:

- Create the proper bundle and move code to the new editor(3) one.
- Start using proper packages for Backbone, jQuery, Carto(DB).js, ...

**Comment by ivanmalagon** [comment]
**tl;dr:** this issue was added for avoiding duplicating a modal #11009 I think it's not worth it for that purpose.

My takeaway after doing some work on this today is that a huge refactor is needed for migrating the code. Some issues that I've found trying to accomplish it:

Moving the `organization` folder to `core/cartodb3` leaves several references to common modules in the old `cartodb` solution. This is a big issue for several reasons:
-  We have to move not only the code from `organizations` folder but also a common dialog  `/common/dialogs/add_group_users`. It's only used apparently in organizations code even if it's on `common` folder. This is not the biggest deal.
- If organizations code is moved to `cartodb3` it includes it in the new build process.The one with pluggable frontends. This is expected and wanted. But it refers to some old common code in `cartodb` codebase. The new build process works for "self contained" codebase but referring to outside code (the old `common` one) doesn't get along with the specs bundling. After refactoring the relative paths of require statements for the runtime bundle, the bundle for specs doesn't find the needed files. I ran into this issue trying to get all the bundles built.
- Organizations code make heavy usage of other packages as stated in the first comment made by @xavijam. In particular, it uses a lot of `cdb` functions related to templates. Some of them can be refactored for using them as in builder, using `require`, but not all of them. There are common templates that should be modified on old `common`code (risk of breaking backwards compatibily) or duplicated in the new `cartodb3` folder (just what we wanted to avoid in first place).

   `cartodb.js` used in organizations is V3 but in Builder is V4. One used from the global namespace, the other using `require`.

    Using views and models from old `common` implies also using helper functions that works with them (like `ViewFactory`) but work using `require` in the very same way. We can use the builder `ViewFactory` but templates in old `cartodb` folder are not the `.tpl` ones but `jst.ejs` ones.
    
    Common elements are used also like `cdb.ui.common.TabPane`, `cdb.admin.Organization.Invites`. Not from the organizations or common folder, but from `cdb` itself. I guess they are replicated in V4 but caution should be taken on this.

   `cdb.god` object is used also in organizations code. In particular, `cdb.god.trigger`. Not sure if the same behaviour is available at Builder.

To sum up, I think this could be done but with a high risk of breaking a lot of little things. This task should be addressed by someone with the knowledge of the whole old `cartodb` code, to know at first glance if something has been already duplicated in `cartodb3` and can be used or we have to reuse / duplicate code. Since we want it to reuse a simple modal... it's not worth it IMHO.

Regards!

**Comment by javierarce** [comment]
Regarding `cdb.god`: we got rid of it in builder, so we'll need to find a workaround for the methods that use it.

**Comment by ivanmalagon** [comment]
Putting it on hold after talking about this with @jorgesancha 

Not the most convenient moment to do it. We can start #11009

**Comment by matallo** [comment]
also, for the brave who takes this one, we're currently running browserify task twice when doing `grunt && grunt dev` and we could tackle that

I'd split it in two (cartodb/cartodb3) as this is the low hanging fruit, maybe there are some further improvements but this would be immediate

**Comment by jorgesancha** [comment]
Removing from the board - deprioritizing

**Comment by noguerol** [comment]
ping
